src/ExpressionEvaluator/CSharp/Test/ResultProvider/FullNameTests.cs

```diff
@@ -909,5 +909,87 @@ public void BackTick_FirstCharacter()
             Verify(GetChildren(children.Single()),
                 EvalResult("x", "0", "int", fullName: null));
         }
+
+        [Fact]
+        public void MangledName_SimplifySynthesizedLocalName()
+        {
+            IDkmClrFullNameProvider2 fullNameProvider = new CSharpFormatter();
+            var inspectionContext = CreateDkmInspectionContext();
+            // The synthesized locals name should just become an empty string.
+            Assert.Equal(string.Empty, fullNameProvider.GetClrNameForLocalVariable(inspectionContext, null, default, default,
+                new DkmClrLocalVariable("CS$<>8__locals0")));
+        }
+
+        [Fact]
+        public void MangledName_SimplifyThisProxyField()
+        {
+            var il = @"
+.class public C
+{
+  .field public object '<>4__this'
+}
+";
+            ImmutableArray<byte> assemblyBytes;
+            ImmutableArray<byte> pdbBytes;
+            CSharpTestBase.EmitILToArray(il, appendDefaultHeader: true, includePdb: false, assemblyBytes: out assemblyBytes, pdbBytes: out pdbBytes);
+            var assembly = ReflectionUtilities.Load(assemblyBytes);
+
+            var fieldToken = assembly.GetType("C").GetFields().First().MetadataToken;
+
+            IDkmClrFullNameProvider2 fullNameProvider = new CSharpFormatter();
+            var inspectionContext = CreateDkmInspectionContext();
+            Assert.Equal(string.Empty,
```

Comment:
>string.Empty [](http://example.com/codeflow?start=25&length=12)

Similar question to above: Where would this empty string appear when debugging?

Reply:
This is used for the Null Reference Exception analysis where we get an identifier for something that was null. In such cases, we expect an empty string for the stashed this (the thing that will be null will be a field in it which is what we care about).
So the metadata name is actually `<>4__this.m_myUserLocalName` was null which becomes
m_myUserLocalName was null since the compiler returns empty for the stashed this.